WordPress/src/main/java/org/wordpress/android/ui/stats/StatsFollowersFragment.java

```diff
@@ -286,19 +287,22 @@ public View getView(int position, View convertView, ViewGroup parent) {
             final FollowDataModel followData = currentRowData.getFollowData();
 
             // entries
-            if (mTopPagerSelectedButtonIndex == 0 && !TextUtils.isEmpty(currentRowData.getURL())) {
-                // WPCOM followers with no empty URL
+            if (mTopPagerSelectedButtonIndex == 0 && !(TextUtils.isEmpty(currentRowData.getURL()) && followData == null)) {
+                // WPCOM followers with no empty URL or empty follow data
 
-                boolean openInReader = true;
+                final int blogID;
                 if (followData == null) {
                     // If follow data is empty, we cannot follow the blog, or access it in the reader.
                     // We need to check if the user is a member of this blog.
                     // If so, we can launch open the reader, otherwise open the blog in the in-app browser.
                     String normURL = normalizeAndRemoveScheme(currentRowData.getURL());
-                    openInReader = dotComUserBlogsURL.contains(normURL);
+                    blogID = userBlogs.containsKey(normURL) ? Integer.parseInt(userBlogs.get(normURL)) : Integer.MIN_VALUE;
```

Comment:
Even when we think it's safe, we should be careful using `parseInt()` since it'll throw `NumberFormatException` on failure. I recommend using `StringUtils.stringToInt()` here instead since it handles the exception and returns a default value on failure, ex:

```
blogID = StringUtils.stringToInt(userBlogs.get(normURL), Integer.MIN_VALUE);
```